backend/functions/lib/adapters/gateways/runtime-state.cosmosdb.mongo.repository.ts

```diff
@@ -0,0 +1,67 @@
+import { ApplicationContext } from '../types/basic';
+import {
+  RuntimeStateRepository,
+  RuntimeState,
+  RuntimeStateDocumentType,
+} from '../../use-cases/gateways.types';
+import { CamsError } from '../../common-errors/cams-error';
+import { DocumentClient } from '../../humble-objects/mongo-humble';
+import { deferClose } from '../../defer-close';
+import QueryBuilder from '../../query/query-builder';
+import { getCamsError } from '../../common-errors/error-utilities';
+import { MongoCollectionAdapter } from './mongo/mongo-adapter';
+
+const MODULE_NAME = 'COSMOS_DB_REPOSITORY_RUNTIME_STATE';
+const COLLECTION_NAME = 'runtime-state';
+
+const { equals } = QueryBuilder;
+
+export class RuntimeStateCosmosMongoDbRepository<T extends RuntimeState>
+  implements RuntimeStateRepository<T>
+{
+  private readonly client: DocumentClient;
+  private readonly databaseName: string;
+
+  constructor(context: ApplicationContext) {
+    const { connectionString, databaseName } = context.config.documentDbConfig;
+    this.databaseName = databaseName;
+    this.client = new DocumentClient(connectionString);
+    deferClose(context, this.client);
+  }
+
+  private getAdapter<T>() {
+    return MongoCollectionAdapter.newAdapter<T>(
+      MODULE_NAME,
+      COLLECTION_NAME,
+      this.databaseName,
+      this.client,
+    );
+  }
```

Comment:
The constructor and getAdapter looks to be implemented the same in every one of these repos.  Can we inherit these from some base class?  I guess getAdapter would have to be protected rather than private though right?

Reply:
Had the same thought. If this was Go we could create a composition. We would have to extend a base class in JS.